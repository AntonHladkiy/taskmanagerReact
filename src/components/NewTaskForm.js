import React, { useState } from 'react';
import {Link} from "react-router-dom";
const NewTaskForm = props => {
    const [task, setTask] = useState(props.initialFormState);

    const handleInputChange = event => {
        const { name, value } = event.target
        setTask({ ...task, [name]: value })
    };

    return (
        <form className={"form-check"} autocomplete="off">
            <label className= "form-check-label">Title</label>
            <input className= "form-control w-25" type="text" name="title" value={task.title} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Description</label>
            <input className= "form-control w-25" type="text" name="description" value={task.position} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Priority</label>
            <input className= "form-control w-25" type="number" name="priority" value={task.priority} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Due Date</label>
            <input className= "form-control w-25" type="date" name="dueDate" value={task.dueDate} onChange={handleInputChange} ></input>
            <Link to ="/"> <button className="btn btn-warning mt-2 mr-2" onClick={()=> {
                if (!task.title || !task.description|| !task.dueDate|| !task.priority) return;
                props.addTask(task)
            }}>Create Task</button></Link>
            <Link to ="/"><button className={"btn btn-danger mt-2"}>Cancel</button></Link>
        </form>
    )
};

export default NewTaskForm;