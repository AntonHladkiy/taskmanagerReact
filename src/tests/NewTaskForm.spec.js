import React from 'react';
import { shallow } from 'enzyme';
import NewTaskForm from '../components/NewTaskForm';
const initialFormState = {
    title:'',
    description:'',
    dueDate:'',
    priority:'',
    done:''
};
let wrapped = shallow(<NewTaskForm initialFormState={initialFormState}></NewTaskForm>);
describe('NewTaskForm', () => {
    it('should render the TaskView Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
    it('renders the NewTaskForm children', () => {
        expect(wrapped.find({ value: '' }).length).toEqual(3)
        expect(wrapped.find({ name: 'dueDate' }).length).toEqual(1)
        expect(wrapped.find({ name: 'description' }).length).toEqual(1)
        expect(wrapped.find({ name: 'priority' }).length).toEqual(1)
    });
});