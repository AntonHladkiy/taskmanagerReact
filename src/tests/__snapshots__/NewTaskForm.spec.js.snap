// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTaskForm should render the TaskView Component correctly 1`] = `
<form
  autoComplete="off"
  className="form-check"
>
  <label
    className="form-check-label"
  >
    Title
  </label>
  <input
    className="form-control w-50"
    name="title"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="form-check-label"
  >
    Description
  </label>
  <input
    className="form-control w-50"
    name="description"
    onChange={[Function]}
    type="text"
  />
  <label
    className="form-check-label"
  >
    Priority
  </label>
  <input
    className="form-control w-50"
    name="priority"
    onChange={[Function]}
    type="number"
    value=""
  />
  <label
    className="form-check-label"
  >
    Due Date
  </label>
  <input
    className="form-control w-50"
    name="dueDate"
    onChange={[Function]}
    type="date"
    value=""
  />
  <Link
    to="/"
  >
     
    <button
      className="btn btn-success mt-2 mr-2"
      onClick={[Function]}
    >
      Create Task
    </button>
  </Link>
  <Link
    to="/"
  >
    <button
      className="btn btn-danger mt-2"
    >
      Cancel
    </button>
  </Link>
</form>
`;
